src/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -171,13 +171,71 @@ public IEnumerable<PackageSource> LoadPackageSources()
             return loadedPackageSources;
         }
 
+        public static void AddDefaultSourcesToRootNuGetConfig()
+        {
+            // Load the root settings (%AppData% one)
+            var settings = Configuration.Settings.LoadDefaultSettings(root: null, configFileName: null, machineWideSettings: null);
+            AddDefaultSourcesToRootNuGetConfigCore(settings);
+        }
+
+        // Internal for unit testing.
+        internal static void AddDefaultSourcesToRootNuGetConfigCore(ISettings settings)
+        {
+            var sourceSettings = settings.GetSettingValues(ConfigurationContants.PackageSources).ToList();
+
+            var settingToMigrate = sourceSettings
+                    .Where(f => f.Key.Equals(NuGetConstants.FeedName, StringComparison.OrdinalIgnoreCase))
+                    .ToArray();
+            var v3Setting = new SettingValue(
+                NuGetConstants.FeedName,
+                NuGetConstants.V3FeedUrl,
+                isMachineWide: false,
+                priority: 1);
+            v3Setting.AdditionalData[ConfigurationContants.ProtocolVersionAttribute] = "3";
+
+            var v2Setting = new SettingValue(
+               NuGetConstants.FeedName,
+               NuGetConstants.V2FeedUrl,
+               isMachineWide: false,
+               priority: 1);
+
+            if (settingToMigrate.Length == 0)
+            {
+                // Potentially empty \ non-existent NuGet.config. Add both v2 and v3 sources
+                sourceSettings.Add(v3Setting);
+                sourceSettings.Add(v2Setting);
+            }
+            else if (settingToMigrate.Any(setting => ReadProtocolVersion(setting) == 3))
+            {
+                // Source already exists for the newest protocol version. Do nothing.
+                return;
+            }
+            else
+            {
+                Debug.Assert(settingToMigrate.Length > 0);
+                var index = sourceSettings.IndexOf(settingToMigrate[0]);
```

Comment:
At this point, you expect that there is only element in 'settingToMigrate'? And, that it is v2 'nuget.org'. If so, I would assert that